test/e2e/proxy_test.go

```diff
@@ -61,10 +64,64 @@ func proxyTestSuite(t *testing.T) {
 	// the deletion of TrustedCAConfigMap, which the former relies on
 	t.Run("Certificate validation", tc.testCertsImport)
 	t.Run("Environment variables validation", tc.testEnvVars)
+
+	t.Run("Request routed through proxy", tc.testProxyRequest)
+	t.Run("Request not proxied", tc.testNoProxyRequest)
+
 	t.Run("Certificate validation removal", tc.testCertsRemoval)
 	t.Run("Environment variables removal validation", tc.testEnvVarRemoval)
 }
 
+// testProxyRequest tests that requests from each Windows node properly go through the proxy
+func (tc *testContext) testProxyRequest(t *testing.T) {
+	clusterProxy, err := tc.client.Config.ConfigV1().Proxies().Get(context.TODO(), "cluster", meta.GetOptions{})
+	if err != nil {
+		require.NoError(t, err)
+	}
+
+	for _, node := range gc.allNodes() {
+		t.Run(node.GetName(), func(t *testing.T) {
+			addr, err := controllers.GetAddress(node.Status.Addresses)
+			require.NoError(t, err, "unable to get node address")
+
+			command := "$response = $(Invoke-WebRequest -UseBasicParsing https://raw.githubusercontent.com/openshift/windows-machine-config-operator/master/README.md); $response.Headers['Via']"
+			out, err := tc.runPowerShellSSHJob("get-webrequest-via-header", command, addr)
+			require.NoErrorf(t, err, "unable to retrieve response 'Via' header")
+
+			viaHeader := strings.TrimSpace(out)
+			fmt.Printf("with proxy Via Header: %s", viaHeader)
+			assert.Truef(t, viaHeader == clusterProxy.Status.HTTPProxy || viaHeader == clusterProxy.Status.HTTPSProxy,
+				"expected response header 'Via' to be either '%s' or '%s' but instead held value: '%s'",
+				clusterProxy.Status.HTTPProxy, clusterProxy.Status.HTTPSProxy, viaHeader)
+		})
```

Comment:
Can there be multiple via headers?
Seems that way looking at [MDN docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Via).
How is that being handled?

Reply:
Good catch. I'll change this to search for a via reader whose value is the cluster-wide proxy
along the lines of 
`$response.Headers["Via"] == clusterProxy.Status.HTTPProxy || $response.Headers["Via"]  == clusterProxy.Status.HTTPSProxy`